File: sparkcql-core/src/test/scala/gov/va/sparkcql/core/StopwatchTest.scala
package gov.va.sparkcql.diagnostic

import gov.va.sparkcql.TestBase

class StopwatchTest extends TestBase {
  "A Stopwatch" should "calculation duration" in {
    var watch = Stopwatch()
    Thread.sleep(10)
    watch.stop()
    assert(watch.lapse.toMillis >= 10)
  }

  it should "format properly" in {
    var watch = Stopwatch()
    Thread.sleep(100)
    watch.stop()
    assert(watch.toString.startsWith("0h:0m:0s:") && watch.toString.endsWith("ms"))
    assert(1 == 1)
  }
}

File: sparkcql-core/src/test/java/gov/va/sparkcql/diagnostic/StopwatchTest.java
package gov.va.sparkcql.diagnostic;

import static org.junit.jupiter.api.Assertions.assertTrue;
import org.junit.jupiter.api.Test;

import gov.va.sparkcql.common.diagnostic.Stopwatch;

public class StopwatchTest {

    @Test
    public void should_calculate_duration() {
        var watch = new Stopwatch();
        watch.sleep(10);
        watch.stop();
        assertTrue(watch.getLapse().toMillis() >= 10);
    }

    @Test
    public void should_format_properly() {
        var watch = new Stopwatch();
        watch.sleep(100);
        watch.stop();
        assertTrue(watch.toString().startsWith("0h:0m:0s:") && watch.toString().endsWith("ms"));
    }
}